File: app/src/test/java/com/turlir/extractors/CommonTest.java
package com.turlir.extractors;

import com.turlir.converter.Member;

import java.util.Iterator;
import java.util.Queue;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public abstract class CommonTest {

    protected static void and(Iterator<Interval> it, String value) {
        assertTrue(it.hasNext());
        assertEquals(it.next(), and(value));
    }

    protected static void or(Iterator<Interval> it, String value) {
        assertTrue(it.hasNext());
        assertEquals(it.next(), or(value));
    }


    //

    protected static void and(Queue<Member> it, Member value) {
        assertFalse(it.isEmpty());
        assertEquals(it.poll(), value);
    }

    protected static void or(Queue<Member> it, Member value) {
        assertFalse(it.isEmpty());
        assertTrue(it.poll() == value);
    }

    //

    protected static Interval or(String value) {
        return new Interval(false, value);
    }

    protected static Interval and(String value) {
        return new Interval(true, value);
    }
}

File: calculator/src/test/java/com/turlir/calculator/extractors/CommonTest.kt
package com.turlir.calculator.extractors

import com.turlir.calculator.converter.Member
import com.turlir.calculator.extractors.Interval
import org.junit.Assert

abstract class CommonTest {

    internal fun and(it: Iterator<Interval>, value: String) {
        Assert.assertTrue(it.hasNext())
        Assert.assertEquals(it.next(), and(value))
    }

    internal fun or(it: Iterator<Interval>, value: String) {
        Assert.assertTrue(it.hasNext())
        Assert.assertEquals(it.next(), or(value))
    }

    //

    protected fun and(it: Iterator<Member>, value: Member) {
        Assert.assertTrue(it.hasNext())
        Assert.assertEquals(it.next(), value)
    }

    protected fun or(it: Iterator<Member>, value: Member) {
        Assert.assertTrue(it.hasNext())
        Assert.assertTrue(it.next() === value)
    }

    //

    protected fun or(value: String): Interval {
        return Interval(false, value)
    }

    protected fun and(value: String): Interval {
        return Interval(true, value)
    }

}
